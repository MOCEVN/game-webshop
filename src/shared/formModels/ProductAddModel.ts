export type ProductAddModel = {
    name: string;
    description: string;
    price: string;
    catagory?: string;
    imageURLs?: string[];
}