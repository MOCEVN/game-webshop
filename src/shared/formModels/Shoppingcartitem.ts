export type Shoppingcartitem = {
    userId: string;
    ammount: string;
    itemId: string;

}