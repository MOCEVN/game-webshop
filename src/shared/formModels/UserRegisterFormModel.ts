export type UserRegisterFormModel = {
    email: string;
    password: string;
    firstName: string;
    lastName: string;
    name: string;
};
