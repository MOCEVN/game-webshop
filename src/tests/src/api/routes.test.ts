import request, { Response } from "supertest";
import app from "../../../api/src/index";
import { expect, it, describe } from "vitest";
import { config } from "dotenv";
import path from "path";
import { UserLoginFormModel } from "@shared/formModels";


config({ path: path.resolve(__dirname,"../../../api/.env")});
config({ path: path.resolve(__dirname,"../../../api/.env.local"), override: true });
process.env.DB_DATABASE = "pb4b2324_toosuutooxii35_test";

describe.concurrent("API endpoints", () => {
    it("route \"/\", get request, success status 200", async () => {
        // Arrange

        // Act
        const res: any = await request(app).get("/");

        // Assert
        expect(res.status).toEqual(200);
    });

    it("route \"/users/login\", email not found, error status 400", async () => {
        // Arrange
        const loginForm: UserLoginFormModel = { email: "xyz@sadfjak.com", password: "2342388" };
        
        // Act
        const res: any = await request(app)
            .post("/users/login")
            .send(loginForm)
            .set("Content-Type", "application/json")
            .set("Accept", "application/json");

        // Assert
        expect(res.status).toEqual(400);
    });

    it("route \"/users/login\", wrong password, error status 400", async () => {
        // Arrange
        const loginForm: UserLoginFormModel = { email: "test@test.nl", password: "wrong-password" };

        // Act
        const res: any = await request(app)
            .post("/users/login")
            .send(loginForm)
            .set("Content-Type", "application/json")
            .set("Accept", "application/json");

        // Assert
        expect(res.status).toEqual(400);
    });

    it("route \"/users/login\", valid login, success status 200", async () => {
        // Arrange
        const loginForm: UserLoginFormModel = { email: "test@test.nl", password: "test" };

        // Act
        const res: any = await request(app)
            .post("/users/login")
            .send(loginForm)
            .set("Content-Type", "application/json")
            .set("Accept", "application/json");

        // Assert
        expect(res.status).toEqual(200);
    });

    it("route \"/store-content/all\", get request, success status 200", async () => {
        // Arrange

        // Act
        const res: any = await request(app)
            .get("/store-content/all")
            .set("Content-Type", "application/json")
            .set("Accept", "application/json");
        
        // Assert
        expect(res.status).toEqual(200);
    });

    it("route \"/store-content/all/:id\", get request, return correct product and status 200", async ({expect}) => {
        // Arrange

        // Act
        const res: Response = await request(app)
            .get("/store-content/all/45")
            .set("Content-Type", "application/json")
            .set("Accept", "application/json");
        
        // Assert
        expect(res.body).toMatchSnapshot();
        expect(res.status).toEqual(200);
    });
});
