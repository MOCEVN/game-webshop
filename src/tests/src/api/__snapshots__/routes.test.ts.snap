// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`API endpoints > route "/store-content/all/:id", get request, return correct product and status 200 1`] = `
{
  "catagory": {
    "description": "this product has no catagory",
    "name": "no catagory",
  },
  "categoryId": 1,
  "description": "Terror Trial is een angstaanjagend huis vol met valstrikken en duistere geheimen, waar wanhopige reizigers gevangen zitten en enkel kunnen ontsnappen door raadsels op te lossen voordat de klok tikt naar hun ondergang.",
  "id": 45,
  "imageURLs": [
    "https://lucastars.hbo-ict.cloud/media/999069f72f4c4c5a98b8e7d25a3ce366/00000006000000000000000000000000.png",
    "https://lucastars.hbo-ict.cloud/media/a37d086f50f14dca8c365448957a2139/00000006000000000000000000000000.png",
    "https://lucastars.hbo-ict.cloud/media/aadfdef650f94bc9a8a99afe21978aa9/00000006000000000000000000000000.png",
    "https://lucastars.hbo-ict.cloud/media/ad5e059aeb314f34b3385bb8b9e91083/00000006000000000000000000000000.png",
  ],
  "price": "0.00",
  "thumbnail": "https://lucastars.hbo-ict.cloud/media/6918722fe0104049b27aa218c692a417/00000006000000000000000000000000.png",
  "title": "Terror Trial",
}
`;
